rdt/hyper_transformer.py

```diff
@@ -171,18 +171,57 @@ def _field_in_data(field, data):
 
     @staticmethod
     def _validate_config(config):
-        sdtypes = config.get('sdtypes', {})
-        transformers = config.get('transformers', {})
-        for column, transformer in transformers.items():
+        if list(config.keys()) != ['sdtypes', 'transformers']:
+            raise Error(
+                'Error: Invalid config. Please provide 2 dictionaries '
+                "named 'sdtypes' and 'transformers'."
+            )
+
+        sdtypes = config['sdtypes']
+        transformers = config['transformers']
+        if set(sdtypes.keys()) != set(transformers.keys()):
+            raise Error(
+                "The column names in the 'sdtypes' dictionary must match the "
+                "column names in the 'transformers' dictionary."
+            )
+
+        unsupported_sdtypes = []
+        supported_sdtypes = HyperTransformer._get_supported_sdtypes()
+        for sdtype in sdtypes.values():
+            if sdtype not in supported_sdtypes:
+                unsupported_sdtypes.append(sdtype)
+
+        if unsupported_sdtypes:
+            raise Error(
+                f'Invalid sdtypes: {unsupported_sdtypes}. If you are trying to use a '
+                'premium sdtype, contact info@sdv.dev about RDT Add-Ons.'
+            )
+
+        invalid_transformers_columns = []
+        mismatched_columns = []
+        for column_name, transformer in transformers.items():
             if transformer is not None:
-                input_sdtype = transformer.get_input_sdtype()
-                sdtype = sdtypes.get(column)
-                if input_sdtype != sdtype:
-                    warnings.warn(
-                        f'You are assigning a {input_sdtype} transformer to a {sdtype} '
-                        f"column ('{column}'). If the transformer doesn't match the "
-                        'sdtype, it may lead to errors.'
-                    )
+                try:
+                    get_transformer_instance(transformer)
+                except ValueError:
+                    invalid_transformers_columns.append(column_name)
```

Comment:
There is a small bug, if we leave it as `ValueError` we are not contemplating other kind of exceptions, if we pass down an `object()` instance this will produce an `AttributeError` : 
`AttributeError: 'object' object has no attribute 'rsplit'`.

Reply:
Can we actually make this change in the `_validate_transformers` method you added in the other issue and just call that method here?